rfcs/0009-code-review.md

```diff
@@ -0,0 +1,192 @@
+     what still needs to be done.
+ - Assume the best intention from review comments
+ - Wait for the tests to pass before you merge.
+ - Tend towards smaller changes. They are easier to understand and
+   faster to review.
+
+
+### Principles for reviewers
+ - Try to comment on the PR in as short an interval is possible (<24
+   hours preferably, excluding weekends).
+   - The comment doesn't have to be the review, but can just be a
+     comment letting the reviewee know when you'll be able to review.
+     This gives them the opportunity to find another reviewer if needed.
+   - If you’re not sure how someone wants their code reviewed or how they
+     prefer to receive feedback, talk to them beforehand.
+ - Point out when something is particularly good or clever.
+   - Reviews often focus on the negative so highlight the good as well
+     when you see it.
+ - Be explicit.
+   - Include screenshots or gifs if they help make your point.
+ - Programming is subjective.
+   - Different approaches are valid.
+   - Ask for clarifications if you need any.
+ - Communicate which ideas you feel strongly about and those you don't.
+
+
+### Technical Considerations
+
+#### Logic
+ - Does the code work?
+   - Does it perform its intended function?
+   - Is the logic correct?
+   - Is the proposed UI accessible?
+ - Does this the change introduce tech debt?
+   - Adding tech debt isn't a bad thing, but we need to make sure we understand
+     why we're adding it and how we're going to contain it.
+   - If adding, are there comments or documentation around the debt so future
+     maintainers can understand the issues?
+ - Does this change introduce any security concerns?
+   - Does it accept or handle user data?
+ - Is all the code easily understood?
+   - Which parts are confusing to you and why?
+   - If you are taking a lot of time to understand the code, perhaps it
+     needs comments or refactoring.
+ - Can the code be simplified?
+   - Write the code you need, not the code you may need.
+ - Does similar functionality already exist in the codebase?
+   - If so, why isn't this functionality reused?
+     (Note: it's okay to not reuse -— but it's worth asking the question.)
+
+#### Comments
+ - Has all commented-out code been deleted?
+   - We can find past code using version control if needed.
+ - Are all functions commented?
+ - Do comments exist to describe the intent of any thorny logic?
+ - Are there comments that no longer make sense?
+
+#### Errors
+ - Can you think of any use case in which the code does not behave as
+   intended?
+ - Are potential errors being caught and logged?
+ - Are third-party utilities used which may return errors?
+ - Is any unusual behavior or edge-case handling described?
+
+#### Tests
+ - Do tests exist, and are they comprehensive, covering good input, bad
+   input and errors?
+ - Are tests checking at the right level?
+ - Do the existing set of tests pass?
```

Comment:
i think this will be automatically checked by Github Actions

Reply:
In general yes, so this will probably be a no-op, but not every project has those actions setup, and there are also possibilities that not all tests run on actions.